bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,155 @@
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+from typing import Literal
+
+from discord import Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+# defining all words in the list of words as a global variable
+with Path("bot/resources/fun/top_1000_used_words.txt").resolve().open(mode="r", encoding="utf-8") as f:
+    ALL_WORDS = [line.strip('\n') for line in f.readlines()]
+
+# defining a list of images that will be used for the game to represent the hangman person
+IMAGES = {tries: image for tries, image in zip(range(6, -1, -1), (
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472355056295946/hangman0.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472756744814613/hangman1.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472808699629578/hangman2.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472862441267230/hangman3.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472950991396864/hangman4.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472999431430204/hangman5.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883473051277226015/hangman6.png",
+)
+)
+}
```

Comment:
Would it really be so hard to just hard-code the numbers?
```suggestion
IMAGES = {
    6: "https://cdn.discordapp.com/attachments/859123972884922418/883472355056295946/hangman0.png",
    5: "https://cdn.discordapp.com/attachments/859123972884922418/883472756744814613/hangman1.png",
    4: "https://cdn.discordapp.com/attachments/859123972884922418/883472808699629578/hangman2.png",
    3: "https://cdn.discordapp.com/attachments/859123972884922418/883472862441267230/hangman3.png",
    2: "https://cdn.discordapp.com/attachments/859123972884922418/883472950991396864/hangman4.png",
    1: "https://cdn.discordapp.com/attachments/859123972884922418/883472999431430204/hangman5.png",
    0: "https://cdn.discordapp.com/attachments/859123972884922418/883473051277226015/hangman6.png",
}
```
The file names are a bit confusing

Reply:
I can add comments to how the filenames work but I would likely have to go into my file system, rename the images, then send it to change the file names, does that sound good with just explaining the filenames?